src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,188 @@
+            return std::wstring(faceName);
+        }
+
+        std::wstring GetFontFamilyName(const wil::com_ptr<IDWriteFontFamily>& fontFamily)
+        {
+            // Retrieve family names.
+            wil::com_ptr<IDWriteLocalizedStrings> familyNames;
+            THROW_IF_FAILED(fontFamily->GetFamilyNames(familyNames.addressof()));
+
+            wchar_t localeNameBuffer[LOCALE_NAME_MAX_LENGTH];
+            const auto localeName = GetUserDefaultLocaleName(localeNameBuffer, LOCALE_NAME_MAX_LENGTH) ? localeNameBuffer : L"en-US";
+
+            UINT32 familyNameIndex;
+            BOOL familyNameExists;
+            if (FAILED(familyNames->FindLocaleName(localeName, &familyNameIndex, &familyNameExists)) || !familyNameExists)
+            {
+                familyNameIndex = 0;
+            }
+
+            UINT32 familyNameLength = 0;
+            THROW_IF_FAILED(familyNames->GetStringLength(familyNameIndex, &familyNameLength));
+            familyNameLength += 1; // Account for the trailing null terminator during allocation.
+
+            wchar_t familyName[32]; // Max size for font name including null terminator is 32.
+            THROW_HR_IF(E_OUTOFMEMORY, familyNameLength > ARRAYSIZE(familyName));
+            THROW_IF_FAILED(familyNames->GetString(0, &familyName[0], familyNameLength));
+            return std::wstring(familyName);
+        }
+    }
+
+    std::vector<FontFamily> GetInstalledFontFamilies()
+    {
+        wil::com_ptr<IDWriteFactory7> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(factory), factory.put_unknown()));
+
+        wil::com_ptr<IDWriteFontCollection> collection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(collection.addressof(), FALSE));
+
+        const auto familyCount = collection->GetFontFamilyCount();
+
+        std::vector<FontFamily> fontFamilies;
+        for (UINT32 index = 0; index < familyCount; index++)
+        {
+            wil::com_ptr<IDWriteFontFamily> family;
+            THROW_IF_FAILED(collection->GetFontFamily(index, family.addressof()));
+
+            std::wstring familyName = GetFontFamilyName(family);
+
+            std::vector<FontFace> fontFaces;
+            const auto& fontCount = family->GetFontCount();
+            for (UINT32 fontIndex = 0; fontIndex < fontCount; fontIndex++)
+            {
+                wil::com_ptr<IDWriteFont> font;
+                THROW_IF_FAILED(family->GetFont(fontIndex, font.addressof()));
+                std::wstring faceName = GetFontFaceName(font);
+
+                wil::com_ptr<IDWriteFontFace> fontFace;
+                THROW_IF_FAILED(font->CreateFontFace(fontFace.addressof()));
+
+                const auto& filePaths = GetFontFilePaths(fontFace);
+
+                FontFace fontFaceEntry;
+                fontFaceEntry.Name = std::wstring(faceName);
+                fontFaceEntry.FilePaths = filePaths;
```

Comment:
Apply `std::move` liberally through this type of function so that you aren't making a copy of something you are about to destroy anyway.